modules/integration_azure-service-bus/detectors-service-bus.tf

```diff
@@ -174,3 +174,43 @@ EOF
     parameterized_body    = var.message_body == "" ? local.rule_body : var.message_body
   }
 }
+
+resource "signalfx_detector" "deadlettered_messages" {
+  name = format("%s %s", local.detector_name_prefix, "Azure Service Bus deadlettered messages")
+
+  authorized_writer_teams = var.authorized_writer_teams
+  teams                   = try(coalescelist(var.teams, var.authorized_writer_teams), null)
+  tags                    = compact(concat(local.common_tags, local.tags, var.extra_tags))
+
+  program_text = <<-EOF
+        base_filter = filter('resource_type', 'Microsoft.ServiceBus/namespaces') and filter('primary_aggregation_type', 'true') and ${module.filtering.signalflow}
+        signal = data('DeadletteredMessages', filter=base_filter)${var.deadlettered_messages_aggregation_function}.publish('signal')
+        detect(when(signal > threshold(${var.deadlettered_messages_threshold_critical}), lasting="${var.deadlettered_messages_timer}")).publish('CRIT')
```

Comment:
do not reverse the major condition in the critical one will lead to duplicated alerts when critical threshold is reached.